src/Tools/ExternalAccess/LiveShare/Projects/RoslynRemoteProjectInfoProvider.cs

```diff
@@ -0,0 +1,114 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.ComponentModel.Composition;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using CustomProtocol = Microsoft.VisualStudio.LanguageServices.LiveShare.CustomProtocol;
+using LSP = Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.LiveShare.Projects
+{
+    //[Export(typeof(IRemoteProjectInfoProvider))]
+    internal class RoslynRemoteProjectInfoProvider : IRemoteProjectInfoProvider
+    {
+        private const string SystemUriSchemeExternal = "vslsexternal";
+
+        private readonly RoslynLSPClientServiceFactory _roslynLSPClientServiceFactory;
+        //private readonly IVsRemoteWorkspaceManager _remoteWorkspaceManager;
+
+        [ImportingConstructor]
+        public RoslynRemoteProjectInfoProvider(RoslynLSPClientServiceFactory roslynLSPClientServiceFactory)//, IVsRemoteWorkspaceManager remoteWorkspaceManager)
+        {
+            _roslynLSPClientServiceFactory = roslynLSPClientServiceFactory ?? throw new ArgumentNullException(nameof(roslynLSPClientServiceFactory));
```

Comment:
so, what is current plan for this? this is not part of LSP. so we need to talk with project system people on this I guess?

Reply:
Yep this is not part of LSP.  Have a work item for it
https://github.com/dotnet/roslyn/projects/45#card-21810235